### DISCUSSION TITLE
React Form Hook with UI5 Web Components React

### DISCUSSION THREAD

**Discussion by apoorvsemwal** [UI5/webcomponents-react#1865] [Q&A] [answered]
Hello,

I am just starting to use React Form Hook with ui5-webcomponents-react and was hoping if somebody could help me with a query I have already posted at
[ReactFormHook-Discussion](https://github.com/react-hook-form/react-hook-form/discussions/6068)

Thanks

**Comment by MarcusNotheis** [comment, marked as answer]
Hey @apoorvsemwal,

thanks for reaching out!

After having a quick look I would assume that you need to update `react-from-hook`: 
It looks like the version you are using (6.15.2) always expects the event to be some kind of the React `SyntheticBaseEvent` because it would always call `event.persist()` which only exists in these kind of events. As the UI5 Web Components (for React) are using "regular" events for e.g. our Button, this code will crash because persist is not a function in the event object.

Newer versions (>= 7.2.0) are supporting other event types as well (see https://github.com/react-hook-form/react-hook-form/pull/4825), so I would assume that upgrading react-form-hook will solve your issue.

Best regards,
Marcus

**Reply by apoorvsemwal** [reply]
Hello @MarcusNotheis 
I really appreciate you taking a look into this and for providing this detailed explanation too.
I tried upgrading the RHF version based on your explanation and it did work out well for me.
Thanks once again. :)
